core/controllers/android_e2e_config.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controller for initializing android specific structures."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from constants import constants
+from core.controllers import base
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import fs_services
+from core.domain import opportunity_services
+from core.domain import question_domain
+from core.domain import question_services
+from core.domain import rights_manager
+from core.domain import skill_domain
+from core.domain import skill_services
+from core.domain import state_domain
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.domain import user_services
+import feconf
+import python_utils
+
+
+class InitializeAndroidTestData(base.BaseHandler):
+    """Handler to initialize android specific structures."""
+
+    def post(self):
+        if constants.DEV_MODE:
+            if topic_services.does_topic_with_name_exist(
+                    'Android test'):
+                topic = topic_fetchers.get_topic_by_name('Android test')
+                topic_rights = topic_fetchers.get_topic_rights(
+                    topic.id, strict=False)
+                if topic_rights.topic_is_published:
+                    raise Exception('The topic is already published.')
+                else:
+                    raise Exception('The topic exists but not published.')
```

Comment:
Added this exception to check if the topic created is published, 
guessing if there might be any errors till topic_services.publish_topic()